### PR TITLE
Signs

### PR DESCRIPTION
**Pull request by RaghuC99**
Added functionality:
1) Auto read signs by walking past them
2) Left clicking signs as a choice
3) Toggle or disable this setting, saves as an option on game reset.
4) Please message me for the updated options GUI menu.

### CODE REVIEW

**Comment by OptimusPi** [comment]
Add *.ipch to your .gitignore and commit your .gitignore. Thanks!

**Comment by OptimusPi** [comment]
same with .log and .pdb please (these may be my fault from recent commits, sorry about that, we'll get it all figured out soon!)

File: StickKnightsOnline-Client/main.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <cstring>
 #include <fstream>
 #include <sstream>
+#include <math.h>
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: Why is this needed for this change?

Also head's up, the naming convention for C++ versions of C libraries is `c"library_name"`

What I mean is, `<string.h>` is a C library and the equivalent C++ library is `<cstring>`

Comment (OptimusPi) [reply StickKnightsOnline-Client/main.cpp]: So, `#include <cmath>` instead of `#include <math.h>` :)

File: StickKnightsOnline-Client/main.cpp [hunk 2]
```diff
@@ -1721,6 +1726,7 @@ void Button::handle_events(int ID)
                                         }
                                     }
                                     //check for interaction with a sign
+
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: remove whitespace

File: StickKnightsOnline-Client/main.cpp [hunk 3]
```diff
@@ -2606,6 +2638,7 @@ void Button::handle_events(int ID)
                            }//end inside inventory click
                           break;
 
+
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: remove whitespace

File: StickKnightsOnline-Client/main.cpp [hunk 4]
```diff
@@ -3834,6 +3868,11 @@ construct_frame()
                DrawImage(0, 244, options_popup);
 
                /* draw little options selectors */
+			   //Sign enabling
+			   if (enableSIGN)
+				   DrawImage(169, 393, option_selector);
+			   else
+				   DrawImage(206, 393, option_selector);
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: take note of how annoying it is to type these "magic numbers" and not be able to configure them except inside main.cpp

Think to the future, how can we read these from a file? How can we make it easier to edit? Something to ponder so we can decide later :)

File: StickKnightsOnline-Client/main.cpp [hunk 5]
```diff
@@ -5009,7 +5048,8 @@ void HandleUI()
 
 					nextPageButton.handle_events(44);
 
-
+					disable_auto_signs_button.handle_events(46);
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: more magic numbers we should refactor (later, and as a whole)

File: StickKnightsOnline-Client/main.cpp [hunk 6]
```diff
@@ -7798,6 +7843,10 @@ int main (int argc, char *argv[])
 
     printf("end of main.\n");
     return 0;
+
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: remove whitespace

File: StickKnightsOnline-Client/main.cpp [hunk 7]
```diff
@@ -8204,12 +8253,53 @@ void physics()
...
 
+			  /*if ( (distance > 175) && (map[current_map]->Sign[i].triggered == true)) {
+				  popup_sign = false;
+			  }*/
 
 
 
+		  }
+	  
+		  float rangeX = (std::abs)(Player[MyID].x - current_sign.x);
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: Possibly could use this math library to simplify some of the code in other areas of the program. I don't mind including it if it's useful and easy to read.

File: StickKnightsOnline-Client/main.cpp [hunk 8]
```diff
@@ -8204,12 +8253,53 @@ void physics()
...
+				  }
+				  if (!current_sign.triggered && !(current_sign.hasBeenClosed)) {
+					  current_sign.triggered = true;
+					  popup_menu = 0;
+					  popup_sign = true;
+					  popup_npc = false;
+				  }
+			  }
 
+			  /*if ( (distance > 175) && (map[current_map]->Sign[i].triggered == true)) {
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: Is there a reason for leaving this here, commented?

File: StickKnightsOnline-Client/main.cpp [hunk 9]
```diff
@@ -8899,3 +8989,5 @@ void Kill()
 	// at the time of writing SDL_Quit() causes a crash upon exit.
 	exit(0);
 }
+
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: Remove whitespace, but if you added a newline to the end of the file that's alright.

File: StickKnightsOnline-Client/main.cpp (hunk 2 above)
Comment (RaghuC99) [reply StickKnightsOnline-Client/main.cpp]: Don't see any on my side.

File: StickKnightsOnline-Client/main.cpp (hunk 8 above)
Comment (RaghuC99) [reply StickKnightsOnline-Client/main.cpp]: Nope. Removed.

**Comment by RaghuC99** [comment]
Made fixes. Submitted another request.

File: StickKnightsOnline-Client/main.cpp [hunk 10]
```diff
@@ -1738,6 +1744,21 @@ void Button::handle_events(int ID)
 											current_sign = map[current_map]->Sign[i];
 											return;
 										}
+
+
+										//bool isColliding = blocked(Player[MyID].x, Player[MyID].y, map[current_map]->Sign[i].x, map[current_map]->Sign[i].y);
+
+
+										bool isColliding = ((Player[MyID].x == map[current_map]->Sign[i].x) && (Player[MyID].y == map[current_map]->Sign[i].y));
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: this looks like a leftover failed attempt. It won't work this way so be sure to remove it

File: StickKnightsOnline-Client/main.cpp [hunk 11]
```diff
@@ -8204,12 +8253,50 @@ void physics()
 
 
 
+	  //Sign reading collision detection
+	  if (enableSIGN) {
```
Comment (OptimusPi) [inline StickKnightsOnline-Client/main.cpp]: Don't do this calculation in Graphics, do it in the Physics thread. Also, don't trigger it every loop when standing still, just when moving.
